okhttp-tests/src/test/java/okhttp3/EventListenerTest.java

```diff
@@ -119,12 +128,55 @@ private void assertSuccessfulEventOrder() throws IOException {
     assertEquals(expectedEvents, listener.recordedEventTypes());
   }
 
+  private void assertBytesReadWritten(RecordingEventListener listener,
+      @Nullable Matcher<Long> requestHeaderLength, @Nullable Matcher<Long> requestBodyBytes,
+      @Nullable Matcher<Long> responseHeaderLength, @Nullable Matcher<Long> responseBodyBytes) {
+
+    if (requestHeaderLength != null) {
+      RequestHeadersEnd responseHeadersEnd = listener.removeUpToEvent(RequestHeadersEnd.class);
+      // TODO implement correct bytes
+      assertThat("request header bytes", responseHeadersEnd.headerLength, requestHeaderLength);
```

Comment:
Just wondering: I hadn't seen these used in other test cases, why the `assertThat`?

Reply:
Backstory is that originally JUnit didn’t have the Hamcrest dependency, and we were trying to avoid adding it!